src/pages/dataSetsWip/form/dataEntryForm/customForm/CustomFormElementsSelector.tsx

```diff
@@ -0,0 +1,184 @@
+                        className={styles.elementButton}
+                        onClick={() => {
+                            insertElement({
+                                type,
+                                id: element.id,
+                                name: element.displayName,
+                                disabled,
+                            })
+                        }}
+                    >
+                        {element.displayName}
+                    </button>
+                </li>
+            ))}
+        </ul>
+    )
+}
+
+export const CustomFormElementsSelector = ({
+    insertElement,
+    previewMode,
+}: {
+    insertElement: ElementSelectorFunction
+    previewMode: boolean
+}) => {
+    const [selectedElementType, setSelectedElementType] = useState<string[]>([])
+    const updateElementTypes = (elementType: string) => {
+        setSelectedElementType((prev) => {
+            const filtered = prev.filter((t) => t !== elementType)
+            return prev.includes(elementType)
+                ? filtered
+                : [...filtered, elementType]
+        })
+    }
+
+    const { loading, elementTypes } = useGetCustomFormElements()
+
+    const [fieldsDisabled, setFieldsDisabled] = useState<boolean>(false)
+    const [filter, setFilter] = useState<string>('')
+
+    if (loading) {
+        return <CircularLoader />
+    }
+    return (
+        <>
+            <SubsectionSpacer>
+                <div className={styles.customFormSubtitle}>
+                    {i18n.t('Select elements')}
+                </div>
+                {previewMode && (
+                    <div className={styles.previewElementsWarning}>
+                        <IconWarning16 />
+                        <div>
+                            {i18n.t(
+                                'Elements cannot be added while previewing the form'
+                            )}
+                        </div>
+                    </div>
+                )}
+            </SubsectionSpacer>
+            <SubsectionSpacer>
+                <InputField
+                    placeholder={i18n.t('Filter elements')}
+                    onChange={(e) => {
+                        setFilter(e?.value ?? '')
+                    }}
+                    value={filter}
+                    className={styles.elementFilter}
+                ></InputField>
+            </SubsectionSpacer>
+
+            <SubsectionSpacer>
+                <Checkbox
+                    checked={fieldsDisabled}
+                    onChange={() => {
+                        setFieldsDisabled((prev) => !prev)
+                    }}
+                    label={i18n.t('Insert grey (disabled) fields')}
+                />
+            </SubsectionSpacer>
+            {elementTypes.map((elementType) => {
+                const selected = selectedElementType.includes(elementType.type)
+                const cleanedFilter = filter
+                    .normalize('NFD')
```

Comment:
this part of the code is to deal with accents. e.g. if I write "donnee", this will also match "Elément de donnée A". Might want to think about filtering more generally though, rather than implementing it differently each time